File: LibeyTechnicalTestAPI/LibeyTechnicalTestDomain/ProvinceAggregate/Domain/Province.cs
﻿namespace LibeyTechnicalTestDomain.ProvinceAggregate.Domain;
public class Province
{
    public string ProvinceCode { get; private set; }
    public string RegionCode { get; private set; }
    public string ProvinceDescription { get; private set; }
    public Province(
        string provinceCode,
        string regionCode,
        string provinceDescription
    ) {
        ProvinceCode = provinceCode;
        RegionCode = regionCode;
        ProvinceDescription = provinceDescription;
    }
}

File: LibeyTechnicalTestWeb/src/app/shared/models/province.ts
export interface Province {
    provinceCode: string;
    regionCode: string;
    provinceDescription: string;
}

export interface ProvinceResponse {
    data: ProvinceResponseItem[];
    success: boolean;
    message: string;
}

export interface ProvinceResponseItem {
    provinceCode: string;
    regionCode: string;
    provinceDescription: string;
}
